src/AbstractEnum.php

```diff
@@ -70,6 +72,28 @@
     }
 
     /**
+     * @param callable|null $callback
+     * @param string|null   $separator
```

Comment:
Please document this. What is the callback called on? What arguments should it have? What does the separator separate?